src/tests/ftest/io/ior_single_server.py

```diff
@@ -20,118 +20,26 @@
-                                self.basepath)
+        Test Description:
+            Test IOR with Single Server config.
 
-    def tearDown(self):
-        try:
-            if self.hostfile_clients is not None:
-                os.remove(self.hostfile_clients)
-            if self.hostfile_servers is not None:
-                os.remove(self.hostfile_servers)
-            if self.pool is not None and self.pool.attached:
-                self.pool.destroy(1)
-        finally:
-            if self.agent_sessions:
-                agent_utils.stop_agent(self.agent_sessions,
-                                       self.hostlist_clients)
-            server_utils.stop_server(hosts=self.hostlist_servers)
-
-    def test_singleserver(self):
-        """
-        Test IOR with Single Server config.
+        Use Cases:
+            Different combinations of 1/64/128 Clients,
+            1K/4K/32K/128K/512K/1M transfer size.
 
         :avocado: tags=ior,singleserver
         """
-
-        # parameters used in pool create
-        createmode = self.params.get("mode", '/run/createtests/createmode/*/')
-        createuid = os.geteuid()
-        creategid = os.getegid()
-        createsetid = self.params.get("setname", '/run/createtests/createset/')
-        createsize = self.params.get("size", '/run/createtests/createsize/')
-        createsvc = self.params.get("svcn", '/run/createtests/createsvc/')
-
-        # ior parameters
-        client_processes = self.params.get("np", '/run/ior/client_processes/*/')
-        iteration = self.params.get("iter", '/run/ior/iteration/')
-        transfer_size = self.params.get("t",
-                                        '/run/ior/transfersize_blocksize/*/')
-        block_size = self.params.get("b", '/run/ior/transfersize_blocksize/*/')
-
-        try:
-            # initialize a python pool object then create the underlying
-            # daos storage
-            self.pool = DaosPool(self.context)
-            self.pool.create(createmode, createuid, creategid,
-                             createsize, createsetid, None, None, createsvc)
-            pool_uuid = self.pool.get_uuid_str()
-            print ("pool_uuid: {}".format(pool_uuid))
-            tmp_rank_list = []
-            svc_list = ""
-            for item in range(createsvc):
-                tmp_rank_list.append(int(self.pool.svc.rl_ranks[item]))
-                svc_list += str(tmp_rank_list[item]) + ":"
-            svc_list = svc_list[:-1]
-
-            ior_utils.run_ior_daos(self.hostfile_clients, self.ior_flags,
-                                   iteration, block_size, transfer_size,
-                                   pool_uuid, svc_list, self.object_class,
-                                   self.basepath, client_processes)
-
-        except (DaosApiError, ior_utils.IorFailed) as excep:
-            self.fail("<Single Server Test FAILED>\n {}".format(excep))
+        self.execute_ior()
```

Comment:
(pylint-no-member) Instance of 'IorSingleServer' has no 'execute_ior' member

Reply:
Fixed